src/App.js

```diff
@@ -30,11 +30,36 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(PLAYER_1);
+  const [winner, setWinner] = useState(null)
+
 
   // Wave 2
   // You will need to create a method to change the square 
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
+  const changeSquares = (id) => {
+
+    // ES6 way to clone an array
+    const newSquares = [...squares];
+
+    // If is not empty then set the value to the current player
+    newSquares.forEach((board) => {
+      board.forEach((square) => {
+        if (square.id === id && square.value === '') {
+          square.value = player;
+          if (player === PLAYER_1) {
+            setPlayer(PLAYER_2);
+          } else {
+            setPlayer(PLAYER_1);
+          };
```

Comment:
I just want to point out you can do this with a ternary.

```suggestion
          setPlayer(  player === PLAYER_1 ? PLAYER_2 : PLAYER_1 );
```